code/events/GenericEvent.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+use League\Event\AbstractEvent;
+
+/**
+ * Class GenericEvent
+ *
+ * This class is a data structure for generic event data, it should be used like this
+ *
+ * $emitter = Injector::inst()->get('EventEmitter');
+ * $emitter->emit(GenericEvent::create('event.name', [
+ *   'random_data' => '1234',
+ *   'some_more' => '5678',
+ * ]));
+ *
+ * Then a listener can subscribe the event like this (_config.php):
+ *
+ * $emitter = Injector::inst()->get('EventEmitter');
+ * $emitter->addListener('event.name', new DomainListener());
+ *
+ * See more on how to use the event system here: http://event.thephpleague.com/2.0/
+ */
+class GenericEvent extends AbstractEvent {
+
+	/**
+	 * @var string
+	 */
+	protected $name;
+
+	/**
+	 * @var array
+	 */
+	protected $data = [];
+
+	/**
+	 * @param string $name - event name
+	 * @param array $data
+	 *
+	 * @return GenericEvent
+	 */
+	public static function create($name, array $data) {
+		return new GenericEvent($name, $data);
```

Comment:
Use Injector->createWithArgs()? I'll merge with or without so i'll leave it up to you.